src/OmniSharp.DotNet/Projects/ProjectSearcher.cs

```diff
@@ -73,19 +93,38 @@ private static IEnumerable<string> FindProjectsThroughGlobalJson(string root)
             GlobalSettings globalSettings;
             if (GlobalSettings.TryGetGlobalSettings(root, out globalSettings))
             {
-                return globalSettings.ProjectSearchPaths
-                                     .Select(searchPath => Path.Combine(globalSettings.DirectoryPath, searchPath))
-                                     .Where(actualPath => Directory.Exists(actualPath))
-                                     .SelectMany(actualPath => Directory.GetDirectories(actualPath))
-                                     .Where(actualPath => File.Exists(Path.Combine(actualPath, Project.FileName)))
-                                     .Select(path => Path.GetFullPath(path))
-                                     .Distinct(StringComparer.OrdinalIgnoreCase)
-                                     .ToList();
-            }
-            else
-            {
-                return Enumerable.Empty<string>();
+                var projectPaths = new SortedSet<string>(StringComparer.OrdinalIgnoreCase);
+                var searchDirectories = new Queue<string>();
+
+                // Look in global.json 'projects' search paths and their immediate children
```

Comment:
For my sanity (because project.json is starting to fade with csproj out soon :stuck_out_tongue_winking_eye:).

`global.json` never supported wild cards, but instead just looked to see if it a any `project.json` in the folder or it's children.  ie the common case was `projects: ["src", "test"]`.